Profiling output respects loglevel.

krk opened the pull request:
### Description

"Profiling started" and other messages are now INFO loglevel and will respect the current loglevel.

### Related issues
https://github.com/async-profiler/async-profiler/issues/960

### Motivation and context
All messages should respect loglevel, "Profiling started" etc. messages did not.

### How has this been tested?
With different `loglevel` parameters:

```
java ... -agentpath: ...,loglevel=info ...
Profiling started
```

```
java ... -agentpath: ...,loglevel=warn ...
```

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

apangin commented:
I'd argue that Profiler output should *not* respect `loglevel`.
Output and logs are two different streams. The former is where profiling results go to; results must be dumped regardless of the log level.

I think a better approach to the problem is to make "Profiler started" a log message, which it essentially is, with an exception of Java API, where this message is a valid response returned from `execute` method.

apangin commented:
When testing, please verify all 3 ways to run profiler:
1. as an agent;
2. using `asprof`;
3. via Java API.

krk commented:
Simplified the approach, introduced a new `LogWriter` class to be used instead of `FileWriter out(STDOUT_FILENO)`.

Tested as an agent, asprof and Java API.

```
❯ asprof start jps
Profiling started

❯ asprof start -L info jps
Profiling started

❯ asprof start -L warn jps
```

apangin commented on src/log.cpp:
```diff
@@ -20,6 +20,9 @@ const char* const Log::LEVEL_NAME[] = {
 LogLevel Log::_level = LOG_INFO;
 
+int Log::fd() {
+    return _file ? fileno(_file) : 0;
```
`_file` must not be NULL, I think.

apangin commented on src/writer.h:
```diff
@@ -34,9 +35,13 @@ class FileWriter : public Writer {
     int _fd;
     char* _buf;
     size_t _size;
+    bool _keepOpen = false;
```
This will issue a compiler warning with older gcc.
We should migrate to a newer C++ standard first, but that's a separate story.

apangin commented on src/writer.h:
```diff
@@ -51,6 +56,16 @@ class FileWriter : public Writer {
+class LogWriter : public FileWriter {
+    LogLevel _logLevel;
+  public:
+    LogWriter(LogLevel logLevel = LOG_INFO) : FileWriter(Log::fd(), true) {
```
This creates an implicit coupling with `Log::_file`, which lives its own life.
Currently, it's not a problem, since `Log` is always closed after the output file, but this dependency is fragile.
Alternatively, `LogWriter` could call `Log::writeRaw` or something.

apangin commented:
After this change, "Profiling started" message goes to `asprof` output. This is intended, but does not look perfect, given that `asprof` prints its own start/stop messages:
```
Profiling started
Profiling for 5 seconds
Done
```
I'm more are more inclined to believe this message is redundant and can be removed entirely except for the Java API call.
